Wire up support for converting integrations-next configs to flow. (#5973)
grafana/agent: docs/sources/flow/getting-started/migrating-from-static.md

}
```

## Integrations Next

> [Integrations next][] configs can be converter or run by adding the `extra-args` flag for [convert][] or `config.extra-args` for [run][].

Reviewer: ```suggestion
You can convert the [integrations next][] configurations, or you can add the `extra-args` flag for [convert][] or `config.extra-args` for [run][] to use the configurations as-is.
```

Author: ```suggestion
You can convert [integrations next][] configurations by adding the `extra-args` flag for [convert][] or `config.extra-args` for [run][].
```

Author: I applied something a little different since I think my intended meaning got jumbled with your improved phrasing.

Reviewer: Looks MUCH better now with the "something a little different" and now it makes sense :-)

{{< code >}}

```static-binary
AGENT_MODE=flow grafana-agent convert --source-format=static --extra-args="-enable-features=integrations-next" --output=<OUTPUT_CONFIG_PATH> <INPUT_CONFIG_PATH>
```

```flow-binary
grafana-agent-flow convert --source-format=static --extra-args="-enable-features=integrations-next" --output=<OUTPUT_CONFIG_PATH> <INPUT_CONFIG_PATH>
```

{{< /code >}}

> 

Reviewer: ```suggestion

 Replace the following:
   * _`<INPUT_CONFIG_PATH>`_: The full path to the [Static][] configuration.
   * _`<OUTPUT_CONFIG_PATH>`_: The full path to output the {{< param "PRODUCT_NAME" >}} configuration.
   
```

## Environment Vars

> You can interpret env variables in your static config using the
> `-config.expand-env` command line flag. This can be passed to [convert][] by
> including `--extra-args="-config.expand-env"` or to [run][]
> by including `--config.extra-args="-config.expand-env"`.

Reviewer: ```suggestion
You can use the `-config.expand-env` command line flag to interpret environment variables in your Grafana Agent Static configuration.
You can pass these flags to [convert][] with `--extra-args="-config.expand-env"` or to [run][] with `--config.extra-args="-config.expand-env"`.
```

> > It is possible to combine `integrations-next` with `expand-env`.
> > For [convert][], `--extra-args="-enable-features=integrations-next -config.expand-env"`

Reviewer: ```suggestion
> It's possible to combine `integrations-next` with `expand-env`.
> For [convert][], you can use `--extra-args="-enable-features=integrations-next -config.expand-env"`
```

## Limitations

Configuration conversion is done on a best-effort basis. {{< param "PRODUCT_ROOT_NAME" >}} will issue warnings or errors where the conversion can't be performed.

After the configuration is converted, review the {{< param "PRODUCT_NAME" >}} configuration file and verify that it's correct before starting to use it in a production environment.

> This following list of limitations is specific to the converter tooling and not {{< param "PRODUCT_NAME" >}}:

Reviewer: ```suggestion
The following list is specific to the convert command and not {{< param "PRODUCT_NAME" >}}:
```

> * The following configuration options are not available for conversion to {{< param "PRODUCT_NAME" >}}: [Traces][] and [Agent Management][].

Reviewer: ```suggestion
* The  [Traces][] and [Agent Management][] configuration options can't be converted to {{< param "PRODUCT_NAME" >}}.
```

Review the following checklist:

> * The following configuration options are not available for conversion to {{< param "PRODUCT_NAME" >}}: [Traces][] and [Agent Management][].

Reviewer 2: Maybe a bit of a scope creep, but should we clarify that traces themselves are supported? It was one of the recent questions we got.

Author: added some wording, LMK what you think
